client/components/d3chart/index.js

```diff
@@ -115,7 +115,7 @@ class D3Chart extends Component {
 			valueType,
 		} = this.props;
 		const { width } = this.state;
-		const calculatedWidth = width || node.offsetWidth;
+		const calculatedWidth = width && width > 0 ? width : node.offsetWidth;
```

Comment:
Is this change required? A `width` of `0` would evaluate to `false`, isn't it?

Reply:
It turns out that in certain circumstances the width comes through as negative, for example `width=-320`, likely from here https://github.com/woocommerce/wc-admin/blob/master/client/components/chart/index.js#L313 and although there is now a catch here https://github.com/woocommerce/wc-admin/blob/master/client/components/chart/index.js#L298 we can't rely on that in other implementations.

Reply:
Makes sense. :+1: